File: BACKEND/Application/RulesTemplate/Responses/RulesTemplateResponse.cs
namespace Application.RulesTemplate.Responses
{
    public class RulesTemplateResponse
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public required string AuthorName { get; set; }

        public int TargetScore { get; set; }
        public bool UseClock { get; set; }
        public int? MatchTimePerPlayerInSeconds { get; set; }
        public int? StartOfTurnDelayPerPlayerInSeconds { get; set; }
        public bool CrawfordRuleEnabled { get; set; }

        public DateTimeOffset CreatedAt { get; set; }
    }
}

File: FRONTEND/backgammon-frontend/src/app/features/rules-template/models/api/responses/rules-template.response.ts
export interface RulesTemplateResponse {
  id: string;

  name: string;
  description?: string | null;
  authorName: string;

  targetScore: number;

  useClock: boolean;

  matchTimePerPlayerInSeconds?: number | null;
  startOfTurnDelayPerPlayerInSeconds?: number | null;

  crawfordRuleEnabled: boolean;

  createdAt: string;
}
